docs/grammar/SolidityLexer.g4

```diff
@@ -0,0 +1,308 @@
+//@doc:inline
+fragment EvenHexDigits: HexCharacter HexCharacter ('_'? HexCharacter HexCharacter)*;
+//@doc:inline
+fragment HexCharacter: [0-9A-Fa-f];
+
+/**
+ * A decimal number literal consists of decimal digits that may be delimited by underscores and
+ * an optional positive or negative exponent.
+ * If the digits contain a decimal point, the literal has fixed point type.
+ */
+DecimalNumber: (DecimalDigits | (DecimalDigits? '.' DecimalDigits)) ([eE] '-'? DecimalDigits)?;
+//@doc:inline
+fragment DecimalDigits: [0-9] ('_'? [0-9])* ;
+
+
+/**
+ * An identifier in solidity has to start with a letter, a dollar-sign or an underscore and
+ * may additionally contain numbers after the first symbol.
+ */
+Identifier: IdentifierStart IdentifierPart*;
+//@doc:inline
+fragment IdentifierStart: [a-zA-Z$_];
+//@doc:inline
+fragment IdentifierPart: [a-zA-Z0-9$_];
+
+WS: [ \t\r\n\u000C]+ -> skip ;
+COMMENT: '/*' .*? '*/' -> channel(HIDDEN) ;
+LINE_COMMENT: '//' ~[\r\n]* -> channel(HIDDEN);
+
+mode AssemblyBlockMode;
+
+//@doc:inline
+AssemblyDialect: '"evmasm"';
+AssemblyLBrace: '{' -> popMode, pushMode(YulMode);
+
+AssemblyBlockWS: [ \t\r\n\u000C]+ -> skip ;
+AssemblyBlockCOMMENT: '/*' .*? '*/' -> channel(HIDDEN) ;
+AssemblyBlockLINE_COMMENT: '//' ~[\r\n]* -> channel(HIDDEN) ;
+
+mode YulMode;
+
+YulBreak: 'break';
+YulCase: 'case';
+YulContinue: 'continue';
+YulDefault: 'default';
+YulFalse: 'false';
+YulFor: 'for';
+YulFunction: 'function';
+YulIf: 'if';
+YulLeave: 'leave';
+YulLet: 'let';
+YulSwitch: 'switch';
+YulTrue: 'true';
+
+/**
+ * Builtin functions in the EVM Yul dialect.
+ */
+YulEVMBuiltin:
+    'stop' | 'add' | 'sub' | 'mul' | 'div' | 'sdiv' | 'mod' | 'smod' | 'exp' | 'not'
+    | 'lt' | 'gt' | 'slt' | 'sgt' | 'eq' | 'iszero' | 'and' | 'or' | 'xor' | 'byte'
+    | 'shl' | 'shr' | 'sar' | 'addmod' | 'mulmod' | 'signextend' | 'keccak256'
+    | 'pop' | 'mload' | 'mstore' | 'mstore8' | 'sload' | 'sstore' | 'msize' | 'gas'
+    | 'address' | 'balance' | 'selfbalance' | 'caller' | 'callvalue' | 'calldataload'
+    | 'calldatasize' | 'calldatacopy' | 'extcodesize' | 'extcodecopy' | 'returndatasize'
+    | 'returndatacopy' | 'extcodehash' | 'create' | 'create2' | 'call' | 'callcode'
+    | 'delegatecall' | 'staticcall' | 'return' | 'revert' | 'selfdestruct' | 'invalid'
+    | 'log0' | 'log1' | 'log2' | 'log3' | 'log4' | 'chainid' | 'origin' | 'gasprice'
+    | 'blockhash' | 'coinbase' | 'timestamp' | 'number' | 'difficulty' | 'gaslimit';
+
+YulLBrace: '{' -> pushMode(YulMode);
+YulRBrace: '}' -> popMode;
+YulLParen: '(';
+YulRParen: ')';
+YulAssign: ':=';
+YulPeriod: '.';
+YulComma: ',';
+YulArrow: '->';
+
+/**
+ * Yul identifiers consist of letters, dollar signs, underscores and numbers, but may not start with a number.
+ */
+YulIdentifier: YulIdentifierStart YulIdentifierPart*;
+//@doc:inline
+fragment YulIdentifierStart: [a-zA-Z$_];
+//@doc:inline
+fragment YulIdentifierPart: [a-zA-Z0-9$_];
```

Comment:
Where's the dot!?

Reply:
In ``yulPath``. There's no dot in identifiers :-).

Reply:
Hidden in `YulPath` in the grammar.

Reply:
For standalone yul I'd do this differently maybe... for inline assembly this makes a lot of sense...

Reply:
Maybe a leave a comment: we do not consider dots here because in inline assembly dots in identifiers are restricted to assignments and expressions.